database/database.go

```diff
@@ -40,6 +57,11 @@ type Database interface {
 	ListWebPushSubscriptions(ctx context.Context, userID, networkID int64) ([]WebPushSubscription, error)
 	StoreWebPushSubscription(ctx context.Context, userID, networkID int64, sub *WebPushSubscription) error
 	DeleteWebPushSubscription(ctx context.Context, id int64) error
+
+	GetMessageLastID(ctx context.Context, networkID int64, name string) (int64, error)
+	StoreMessage(ctx context.Context, networkID int64, name string, msg *irc.Message) (int64, error)
+	ListMessageLastPerTarget(ctx context.Context, networkID int64, options MessageOptions) ([]MessageTarget, error)
+	ListMessages(ctx context.Context, networkID int64, name string, options MessageOptions) ([]*irc.Message, error)
```

Comment:
Nit: can we pass the options as pointers here? There are quite a few fields in there.